Framework/PythonInterface/mantid/kernel/src/Registry/PropertyWithValueFactory.cpp

```diff
@@ -138,6 +150,40 @@ PropertyWithValueFactory::create(const std::string &name,
   return create(name, defaultValue, validator, direction);
 }
 
+/**
+ * Creates a TimeSeriesProperty<Type> instance from the given information.
+ * The python type is mapped to a C type
+ * @param name :: The name of the property
+ * @param defaultValue :: A default value for this property.
+ * @returns A pointer to a new Property object
+ */
+std::unique_ptr<Mantid::Kernel::Property>
+PropertyWithValueFactory::createTimeSeries(
+    const std::string &name, const boost::python::object &defaultValue) {
+
+  // Take in the object and try to convert it into a list
+  const boost::python::list pyList = extract<boost::python::list>(defaultValue);
+
+  // Use a PyObject pointer to determine the type stored in the list
+  auto object = boost::python::object(pyList[0]).ptr();
+
+  // Decide which kind of TimeSeriesProperty to return
+  if (PyBool_Check(object) == 1) {
```

Comment:
Don't use `PyBool_Check` use: 
```cpp
boost::python::extract<bool>(object).check()
```

Reply:
Also why not add a using statement for boost::python?

Reply:
I tried extracting with boost first and I think the issue was that the `.check()` function accepted 1 as a Boolean value. So if a list of integers was given e.g. [1,2,3,4,5] then a `BoolTimeSeriesProperty` was returned.

Reply:
Ah i see. Instead I guess we want to check that we can convert to a std::vector. I think there might be something in the `Converters` namespace that can do this.

Reply:
You could always expect a `boost::python::list` and then check size, then check the type of the first item. Here's an example: https://stackoverflow.com/a/3761526

Reply:
Okay I'll give that a go!